packages/calling-stateful-client/src/CallContext.ts

```diff
@@ -160,6 +160,28 @@ export class CallContext {
     });
   }
 
+  /* @conditional-compile-remove(PSTN-calls) */
+  public setCallHold(callId: string): void {
+    const latestCallId = this._callIdHistory.latestCallId(callId);
+    this.modifyState((draft: CallClientState) => {
+      const call = draft.calls[latestCallId];
+      if (call && call.state !== 'RemoteHold') {
```

Comment:
Why can't I put a call on `LocalHold` if it is on `RemoteHold`?

Reply:
Also, I would assume a call can't be put on hold if it is in transient states like `Ringing` `Connecting` etc.

Reply:
I will do some more digging into this but this I think is a behavior that would mirror when on a phone call between 2 phones. If someone puts me on hold I don't know that I should be able to do anything in the call other than hang up. 

Will update for the transient states. Was just thinking that the button wouldn't be present when the call is connecting but Contoso could have the control bar up when they are connecting to a call in their component app.